src/main/java/htsjdk/samtools/SAMFileHeader.java

```diff
@@ -253,7 +255,12 @@ public SortOrder getSortOrder() {
         if (so == null || so.equals("unknown")) {
             return SortOrder.unsorted;
         }
-        return SortOrder.valueOf((String) so);
+        try {
+            return SortOrder.valueOf(so);
+        } catch (IllegalArgumentException e) {
+            log.warn("Found non conforming header SO tag: "+ so + ". Treating as 'unknown'.");
```

Comment:
I don't like changing the header, as that might have side-effects (the header might be used later to write a file.)

I'll try to cache the resulting enum.